internal/jimm/bootstrap/types.go

```diff
@@ -0,0 +1,40 @@
+// Copyright 2025 Canonical.
+
+package bootstrap
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/canonical/jimm/v3/internal/errors"
+)
+
+// BootstrapParams defines the parameters required for bootstrapping a JIMM controller.
+type BootstrapParams struct {
+	ControllerName string
+	CloudName      string
+	CloudRegion    string
+	AgentVersion   string
+	TimeoutSeconds int
+}
+
+// Validate checks if the BootstrapParams are valid.
+func (p BootstrapParams) validate() error {
+	var msgs []string
```

Comment:
while this is a very user friendly approach, the resulting error message might be confusing (as it returns a multi-line message).. i would prefer to just return a simple error message in each of the if statements.. but that's just me.. try try try again, until you succeed :)